FunctionalArray.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FunctionalArray.swift
+//  SwiftTrainingExercise
+//
+//  Created by Ahmed Ibrahim on 20/06/2023.
+//
+
+import Foundation
+
+class FunctionalArray<Element> where Element: Equatable & Comparable & Hashable {
+  
+  var elementsArray: [Element] = [] // how to set this as private only for subclass and superclass so its not accessable ?
+  private(set) var maxSize = Int.max;
```

Comment:
why is private set

Reply:
So that no one can access it and change it manually, we already have a resize() function that should be used when we want to make resize, otherwise we need to call the resize function in the didSet of maxSize